tendrl/commons/objects/cluster/atoms/unmanage_cluster/__init__.py

```diff
@@ -0,0 +1,141 @@
+                    self.parameters["Node[]"] = json.loads(_node_ids)
+
+                except etcd.EtcdKeyNotFound:
+                    _cluster = NS.tendrl.objects.Cluster(
+                        integration_id=NS.tendrl_context.integration_id).load()
+                    _cluster.status = ""
+                    _cluster.current_job['status'] = 'failed'
+                    _cluster.save()
+                    raise AtomExecutionFailedError(
+                        "Could not execute UnmanageCluster atom - "
+                        "Could not load Nodelist from etcd")
+
+            node_list = self.parameters['Node[]']
+            if len(node_list) > 1:
+                # This is the master node for this flow
+                # Lock nodes
+                flow_utils.acquire_node_lock(self.parameters)
+                NS.tendrl_context = NS.tendrl_context.load()
+                # Creating child jobs for nodes to unamange themselves
+                for node in node_list:
+                    if NS.node_context.node_id != node:
+                        new_params = self.parameters.copy()
+                        new_params['Node[]'] = [node]
+                        # create same flow for each node in node list except
+                        #  $this
+                        payload = {"tags": ["tendrl/node_%s" % node],
+                                   "run": "tendrl.flows.UnmanageCluster",
+                                   "status": "new",
+                                   "parameters": new_params,
+                                   "parent": self.parameters['job_id'],
+                                   "type": "node"
+                                   }
+                        _job_id = str(uuid.uuid4())
+                        NS.tendrl.objects.Job(
+                            job_id=_job_id,
+                            status="new",
+                            payload=payload
+                        ).save()
+                        logger.log(
+                            "info",
+                            NS.publisher_id,
+                            {"message": "UnmanageCluster %s (jobID: %s) :"
+                                        "removing host %s" %
+                                        (_cluster.short_name, _job_id, node)},
+                            job_id=self.parameters['job_id'],
+                            flow_id=self.parameters['flow_id']
+                        )
+                logger.log(
+                    "info",
+                    NS.publisher_id,
+                    {"message": "UnmanageCluster %s waiting for hosts %s "
+                                "to be unmanaged"
+                                % (_cluster.short_name, node_list)},
+                    job_id=self.parameters['job_id'],
+                    flow_id=self.parameters['flow_id']
+                )
+                loop_count = 0
+                # Wait for (no of nodes) * 6 minutes for unmanage to complete
```

Comment:
Do we really need to wait for 6 mins per node for un-manage to complete? Its not an import where packages installation etc happen. So I feel a couple of mins would more more than required.

Reply:
Sure. Will reduce the time